pennylane/gradients/general_shift_rules.py

```diff
@@ -170,42 +166,38 @@ def _get_shift_rule(frequencies, shifts=None):
         )
 
     else:  # non-equidistant case
-        sin_matr = -4 * np.sin(np.outer(shifts, frequencies))
-        det_sin_matr = np.linalg.det(sin_matr)
-        if abs(det_sin_matr) < 1e-6:
+        sin_matrix = -4 * np.sin(np.outer(shifts, frequencies))
+        det_sin_matrix = np.linalg.det(sin_matrix)
+        if abs(det_sin_matrix) < 1e-6:
             warnings.warn(
-                f"Solving linear problem with near zero determinant ({det_sin_matr}) "
+                f"Solving linear problem with near zero determinant ({det_sin_matrix}) "
                 "may give unstable results for the parameter shift rules."
             )
 
-        coeffs = -2 * np.linalg.solve(sin_matr.T, frequencies)
+        coeffs = -2 * np.linalg.solve(sin_matrix.T, frequencies)
 
     coeffs = np.concatenate((coeffs, -coeffs))
     shifts = np.concatenate((shifts, -shifts))  # pylint: disable=invalid-unary-operand-type
-    return np.stack([coeffs, shifts])
+    return np.stack([coeffs, shifts]).T
 
 
 def _iterate_shift_rule_with_multipliers(rule, order, period):
     r"""Helper method to repeat a shift rule that includes multipliers multiple
     times along the same parameter axis for higher-order derivatives."""
     combined_rules = []
 
-    # TODO: Optimization: Only iterate over half of the combinations and
-    # double their coefficients (does this work with multipliers?)

```

Comment:
Was this TODO removed because it is no longer applicable with multipliers?

Reply:
Yes. At least I was not able to find a way of achieving this, so the TODO is not really a clear TODO anymore :)